File: inst/rstudio-spcs/smoke_test.py
#!/opt/conda/envs/snowflake_ml/bin/python
"""Smoke test: Snowpark + SPCS OAuth in RStudio SPCS container.

Run from RStudio Terminal:
  python3 ~/smoke_test.py
"""

from __future__ import annotations

import os
import sys


def _require(path: str) -> str:
    if not os.path.exists(path):
        print(f"FAIL: missing {path}", file=sys.stderr)
        sys.exit(1)
    return path


def main() -> None:
    print("=== Python / Snowpark smoke test ===")
    token_path = _require("/snowflake/session/token")
    host = os.environ.get("SNOWFLAKE_HOST", "")
    if not host:
        print("FAIL: SNOWFLAKE_HOST not set", file=sys.stderr)
        sys.exit(1)

    print(f"SNOWFLAKE_HOST: {host}")
    print(f"Token file: {token_path}")

    from snowflake.snowpark import Session

    token = open(token_path, encoding="utf-8").read().strip()
    session = (
        Session.builder.configs(
            {
                "host": host,
                "account": os.environ["SNOWFLAKE_ACCOUNT"],
                "authenticator": "oauth",
                "token": token,
                "warehouse": os.environ.get("SNOWFLAKE_WAREHOUSE", ""),
                "database": os.environ.get("SNOWFLAKE_DATABASE", ""),
                "schema": os.environ.get("SNOWFLAKE_SCHEMA", ""),
                "role": os.environ.get("SNOWFLAKE_ROLE", ""),
            }
        ).create()
    )

    rows = session.sql(
        "SELECT CURRENT_USER() AS u, CURRENT_WAREHOUSE() AS wh, CURRENT_ROLE() AS r"
    ).collect()
    print(rows[0].as_dict() if rows else "FAIL: no rows")
    print("=== PASS ===")


if __name__ == "__main__":
    main()

File: inst/rstudio-spcs/smoke_test.R
# Smoke test for RStudio on SPCS — RSnowflake + snowflakeR + reticulate
# Run: source("~/smoke_test.R")

message("=== RStudio SPCS smoke test ===")

stopifnot(
  "SPCS token or SNOWFLAKE_HOST required" =
    nzchar(Sys.getenv("SNOWFLAKE_HOST", "")) ||
    file.exists("/snowflake/session/token")
)

cat("SNOWFLAKE_HOST:", Sys.getenv("SNOWFLAKE_HOST"), "\n")
cat("SNOWFLAKE_WAREHOUSE:", Sys.getenv("SNOWFLAKE_WAREHOUSE", "<unset>"), "\n")
cat("Token file exists:", file.exists("/snowflake/session/token"), "\n")

message("\n=== RSnowflake ===")
library(DBI)
library(RSnowflake)

con <- dbConnect(Snowflake())
on.exit(try(dbDisconnect(con), silent = TRUE), add = TRUE)

who <- dbGetQuery(con, "
  SELECT
    CURRENT_USER()   AS user,
    CURRENT_ROLE()   AS role,
    CURRENT_WAREHOUSE() AS warehouse,
    CURRENT_VERSION() AS version
")
print(who)

sample <- head(dbGetQuery(con, "
  SELECT C_CUSTKEY, C_NAME
  FROM SNOWFLAKE_SAMPLE_DATA.TPCH_SF1.CUSTOMER
  LIMIT 3
"))
print(sample)
message("RSnowflake: OK")

message("\n=== snowflakeR (SPCS Snowpark) ===")
stopifnot(requireNamespace("snowflakeR", quietly = TRUE))
stopifnot(requireNamespace("reticulate", quietly = TRUE))

source("~/spcs_helpers.R")
ml <- sfr_connect_spcs()

result <- ml$session$sql(
  "SELECT CURRENT_USER() AS u, CURRENT_WAREHOUSE() AS wh"
)$collect()
print(reticulate::py_to_r(result))
message("snowflakeR: OK")

message("\n=== ALL PASS ===")
